src/hapi/pipelines/database/refugees.py

```diff
@@ -0,0 +1,99 @@
+"""Functions specific to the refugees theme."""
+
+from logging import getLogger
+from typing import Dict
+
+from hapi_schema.db_refugees import DBRefugees
+from hdx.utilities.dateparse import parse_date
+from hdx.utilities.dictandlist import dict_of_lists_add
+from sqlalchemy.orm import Session
+
+from ..utilities.parse_tags import (
+    get_gender_and_age_range,
+    get_min_and_max_age,
+)
+from . import admins
+from .base_uploader import BaseUploader
+from .metadata import Metadata
+
+logger = getLogger(__name__)
+
+
+class Refugees(BaseUploader):
+    def __init__(
+        self,
+        session: Session,
+        metadata: Metadata,
+        locations: admins.Locations,
+        results: Dict,
+    ):
+        super().__init__(session)
+        self._metadata = metadata
+        self._locations = locations
+        self._results = results
+
+    def populate(self):
+        logger.info("Populating refugees table")
+        for dataset in self._results.values():
+            for admin_level, admin_results in dataset["results"].items():
+                resource_id = admin_results["hapi_resource_metadata"]["hdx_id"]
+                results = admin_results["values"]
+                hxl_tags = admin_results["headers"][1]
+                date_i = hxl_tags.index("#date+year")
+                asylum_location_i = hxl_tags.index("#country+code+asylum")
+                population_group_i = hxl_tags.index(
+                    "#indicator+population_type"
+                )
+
+                for hxl_index, hxl_tag in enumerate(hxl_tags):
+                    if hxl_index in [
+                        date_i,
+                        asylum_location_i,
+                        population_group_i,
+                    ]:
+                        continue
+                    gender, age_range = get_gender_and_age_range(hxl_tag)
+                    min_age, max_age = get_min_and_max_age(age_range)
+                    rows = {}
+                    for origin_location in results[hxl_index]:
+                        values = results[hxl_index][origin_location]
+                        for i, population in enumerate(values):
+                            date = results[date_i][origin_location][i]
+                            asylum_location = results[asylum_location_i][
+                                origin_location
+                            ][i]
+                            population_group = results[population_group_i][
+                                origin_location
+                            ][i]
+                            dict_of_lists_add(
```

Comment:
I guess this is needed because you have to sum over the locations?

Reply:
I realized I didn't leave any comments, I can add at least one to explain this.